### PR TITLE
[GEOT 6223] - update GeoJSONStore to Jackson

### PR DESCRIPTION
**Pull request by ianturton**

### CODE REVIEW

File: modules/unsupported/geojsonstore/pom.xml [hunk 1]
```diff
@@ -28,14 +24,29 @@
...
 			<version>${project.version}</version>
 			<scope>test</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.bedatadriven</groupId>
```
Comment (mprins) [inline modules/unsupported/geojsonstore/pom.xml]: Hi @ianturton You may want to look into using the forked version from the graphhopper team that has been upgraded wr/ JTS 

eg. 
```xml
<dependency>
  <groupId>com.graphhopper.external</groupId>
  <artifactId>jackson-datatype-jts</artifactId>
  <version>[latest]</version>
</dependency> 
```

see: https://github.com/graphhopper/jackson-datatype-jts

bedatadriven/jackson-datatype-jts seems to have been [abandoned](https://github.com/bedatadriven/jackson-datatype-jts/pull/18#issuecomment-470858490)

**Comment by aaime** [comment]
Gave the build a spin on Windows, there are legit failures:

``
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.geotools.data.geojson.GeoJSONDataStoreFactoryTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.641 s - in org.geotools.data.geojson.GeoJSONDataStoreFactoryTest
[INFO] Running org.geotools.data.geojson.GeoJSONDataStoreTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.031 s - in org.geotools.data.geojson.GeoJSONDataStoreTest
[INFO] Running org.geotools.data.geojson.GeoJSONReaderTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.geotools.data.geojson.GeoJSONReaderTest
[INFO] Running org.geotools.data.geojson.GeoJSONWriteTest
[ERROR] Tests run: 8, Failures: 0, Errors: 7, Skipped: 0, Time elapsed: 0.281 s <<< FAILURE! - in org.geotools.data.geojson.GeoJSONWriteTest
[ERROR] transactionExample(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0.219 s  <<< ERROR!
java.nio.file.FileSystemException:
C:\Users\geowolf\AppData\Local\Temp\example1620703794368212058\locations.json: Impossibile accedere al file. Il file è utilizzato da un altro processo.

        at org.geotools.data.geojson.GeoJSONWriteTest.transactionExample(GeoJSONWriteTest.java:216)

[ERROR] transactionExample(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0.219 s  <<< ERROR!
java.io.IOException: could not delete: C:\Users\geowolf\AppData\Local\Temp\example1620703794368212058\locations.json
        at org.geotools.data.geojson.GeoJSONWriteTest.removeTemporaryLocations(GeoJSONWriteTest.java:120)

[ERROR] replaceAll(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0.016 s  <<< ERROR!
java.nio.file.FileSystemException:
C:\Users\geowolf\AppData\Local\Temp\example580086585923652400\locations.json: Impossibile accedere al file. Il file è utilizzato da un altro processo.

        at org.geotools.data.geojson.GeoJSONWriteTest.replaceAll(GeoJSONWriteTest.java:332)

[ERROR] replaceAll(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0.016 s  <<< ERROR!
java.io.IOException: could not delete: C:\Users\geowolf\AppData\Local\Temp\example580086585923652400\locations.json
        at org.geotools.data.geojson.GeoJSONWriteTest.removeTemporaryLocations(GeoJSONWriteTest.java:120)

[ERROR] copyContent(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0 s  <<< ERROR!
java.nio.file.FileSystemException:
C:\Users\geowolf\AppData\Local\Temp\example3000391656845902549\duplicate.geojson: Impossibile accedere al file. Il file è utilizzato da un altro processo.

        at org.geotools.data.geojson.GeoJSONWriteTest.copyContent(GeoJSONWriteTest.java:386)

[ERROR] copyContent(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0 s  <<< ERROR!
java.io.IOException: could not delete: C:\Users\geowolf\AppData\Local\Temp\example3000391656845902549\duplicate.geojson
        at org.geotools.data.geojson.GeoJSONWriteTest.removeTemporaryLocations(GeoJSONWriteTest.java:120)

[ERROR] removeAllExample(org.geotools.data.geojson.GeoJSONWriteTest)  Time elapsed: 0.016 s  <<< ERROR!
java.io.IOException: could not delete: C:\Users\geowolf\AppData\Local\Temp\example3081714481813572489\locations.json
        at org.geotools.data.geojson.GeoJSONWriteTest.removeTemporaryLocations(GeoJSONWriteTest.java:120)
``

**Comment by aaime** [comment]
HI @ianturton just pushed a few changes that make the build pass on my Windows laptop (when doing PRs one by default gives permissions to project owners to push to the PR branch directly).

**Comment by ianturton** [comment]
Thanks @aaime - that was doing my head in! I'm going to try @mprins suggestion to get rid of the JTS Json writer so it's all Jackson later today.

**Pull request merged**